variables.tf

```diff
@@ -1,76 +1,113 @@
+  - `name` - The name of the route table entry.
+  - `address_prefix` - The address prefix to match for this route table entry.
+  - `next_hop_type` - The type of the next hop. Possible values include `Internet`, `VirtualAppliance`, `VirtualNetworkGateway`, `VnetLocal`, `None`.
+  - `has_bgp_override` - Should the BGP override be enabled for this route table entry? Default `false`.
+  - `next_hop_ip_address` - The IP address of the next hop. Required if `next_hop_type` is `VirtualAppliance`.
+
+- `subnets` - A map of subnets to create in the virtual network. The value is an object with the following fields:
+  - `address_prefixes` - The IPv4 address prefixes to use for the subnet in CIDR format.
+  - `nat_gateway` - (Optional) An object with the following fields:
+    - `id` - The ID of the NAT Gateway which should be associated with the Subnet. Changing this forces a new resource to be created.
+  - `network_security_group` - (Optional) An object with the following fields:
+    - `id` - The ID of the Network Security Group which should be associated with the Subnet. Changing this forces a new association to be created.
+  - `private_endpoint_network_policies_enabled` - (Optional) Enable or Disable network policies for the private endpoint on the subnet. Setting this to true will Enable the policy and setting this to false will Disable the policy. Defaults to true.
+  - `private_link_service_network_policies_enabled` - (Optional) Enable or Disable network policies for the private link service on the subnet. Setting this to true will Enable the policy and setting this to false will Disable the policy. Defaults to true.
+  - `assign_generated_route_table` - (Optional) Should the Route Table generated by this module be associated with this Subnet? Default `true`. Cannot be used with `external_route_table_id`.
+  - `external_route_table_id` - (Optional) The ID of the Route Table which should be associated with the Subnet. Changing this forces a new association to be created. Cannot be used with `assign_generated_route_table`.
+  - `service_endpoints` - (Optional) The list of Service endpoints to associate with the subnet.
+  - `service_endpoint_policy_ids` - (Optional) The list of Service Endpoint Policy IDs to associate with the subnet.
+  - `service_endpoint_policy_assignment_enabled` - (Optional) Should the Service Endpoint Policy be assigned to the subnet? Default `true`.
+  - `delegation` - (Optional) An object with the following fields:
+    - `name` - The name of the delegation.
+    - `service_delegation` - An object with the following fields:
+      - `name` - The name of the service delegation.
+      - `actions` - A list of actions that should be delegated, the list is specific to the service being delegated.
+
+DESCRIPTION
   type = map(object({
     name                = string
-    address_spaces      = list(string)
+    address_space       = list(string)
```

Comment:
Will `address_space` be considered as one string?

Reply:
I was following the AzureRM precedent, though I have just noticed that subnets input is also using the plural.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_network.html#address_space

Reply:
back to plural